vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -35,6 +47,21 @@ def _filter_isin(series: pd.Series, value: MultiValueType) -> pd.Series:
     return series.isin(value)
 
 
+def _filter_date_between(series: pd.Series, value: List[str]) -> pd.Series:
+    series = pd.to_datetime(series)
+    start_date, end_date = value
+
+    return series.between(start_date, end_date, inclusive="both")
+
+
+def _filter_date_isin(series: pd.Series, value: MultiValueType) -> pd.Series:
+    series = pd.to_datetime(series)
+    return series.isin(value)
```

Comment:
I still think that we should not need these extra functions, but should be able to re-use the existing ones. To get that to work, I think we have to convert the relevant column in the data set **before it gets into the filter function**. To illustrate what I mean with your current dev example:

This should work without any errors, but currently it breaks: 
I think we should encode that logic in that if the data column can be converted to a datetime object, then we do that automatically and apply the right default selector (DatePicker). Currently this is not the case, as we don't convert the data column automatically and this is also the reason why you currently need these two extra functions here. 

If we would convert the dtype for the column smartly as well, the above example should work out of the box (using the Datepicker automatically) and you might not need these two extra functions

```
page_1 = vm.Page(
    title="Datepicker page",
    components=[vm.Table(id="table", figure=dash_data_table(data_frame=data))],
    controls=[
        vm.Filter(column="DATE_TIME")
    ],
)
```